src/components/input/range/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { Component, Event, EventEmitter, h, Host, Listen, Prop, Watch } from "@stencil/core"
+
+@Component({
+	tag: "smoothly-input-range",
+	styleUrl: "style.css",
+	shadow: true,
+})
+export class SmoothlyInputRange {
+	slide?: HTMLDivElement
+	slider?: HTMLDivElement
+	isDragging = false
+	private set ratio(value: number) {
```

Comment:
i get a feeling most of these can go if most of them are used to compute values that is then used for inline styles and some of those inline styles can be replace with "normal" css rules